Src/LinearSolvers/MLMG/AMReX_MLNodeLaplacian.cpp

```diff
@@ -323,55 +323,133 @@ MLNodeLaplacian::interpolation (int amrlev, int fmglev, MultiFab& fine, const Mu
         AMREX_ALWAYS_ASSERT(regular_coarsening);
     }
 
-#ifdef AMREX_USE_OMP
-#pragma omp parallel if (Gpu::notInLaunchRegion())
-#endif
-    for (MFIter mfi(fine, TilingIfNotGPU()); mfi.isValid(); ++mfi)
-    {
-        Box const& bx = mfi.tilebox();
-        Array4<Real> const& ffab = fine.array(mfi);
-        Array4<Real const> const& cfab = cmf->const_array(mfi);
-        Array4<int const> const& mfab = dmsk.const_array(mfi);
+    auto fine_ma = fine.arrays();
+    auto crse_ma = cmf->const_arrays();
+    auto msk_ma = dmsk.const_arrays();
+
+    if (Gpu::inLaunchRegion()) {
+        IntVect ng( 0);
         if (m_coarsening_strategy == CoarseningStrategy::RAP)
         {
-            Array4<Real const> const& stfab = stencil->const_array(mfi);
-            AMREX_HOST_DEVICE_PARALLEL_FOR_3D(bx, i, j, k,
+            auto sten_ma = stencil->const_arrays();
+            experimental::ParallelFor(fine, ng,[=] AMREX_GPU_DEVICE(int box_no, int i, int j, int k) noexcept
             {
-                mlndlap_interpadd_rap(i,j,k,ffab,cfab,stfab,mfab);
+                mlndlap_interpadd_rap(box_no, i, j, k, fine_ma, crse_ma, sten_ma, msk_ma);
```

Comment:
Could we make some changes here and other places?
```
mlndlap_interpadd_rap(i, j, k, fine_ma[box_no], crse_ma[box_no], sten_ma[box_no], msk_ma[box_no]);
```

Reply:
If we do it this way, we no longer need to modify `AMReX_MLNodeLap_?D_K.H` add new versions of `mlndlap_interpadd_*`.

Reply:
I see what you want. Hopefully the latest commit addresses your concerns.